File: List/Predko/LinkedList.java
package org.example.Predko_List.Predko;

class LinkedList<T> {
    Node<T> head;

    LinkedList() {
        this.head = null;
    }

    public boolean isEmpty() {
        return head == null;
    }

    public void append(T data) {
        Node<T> newNode = new Node<>(data);
        if (head == null) {
            head = newNode;
            return;
        }
        Node<T> current = head;
        while (current.next != null) {
            current = current.next;
        }
        current.next = newNode;
    }

    public void appendFirst(T data) {
        Node<T> newNode = new Node<>(data);
        newNode.next = head;
        head = newNode;
    }

    public void appendAtIndex(int index, T data) {
        if (index < 0)
            throw new IndexOutOfBoundsException("Index out of bounds");

        if (index == 0) {
            appendFirst(data);
            return;
        }

        Node<T> newNode = new Node<>(data);
        Node<T> current = head;
        Node<T> prev = null;
        int count = 0;

        while (current != null && count != index) {
            prev = current;
            current = current.next;
            count++;
        }

        if (current == null && count != index)
            throw new IndexOutOfBoundsException("Index out of bounds");

        newNode.next = current;
        prev.next = newNode;
    }

    public void deleteLast() {
        if (head == null || head.next == null) {
            head = null;
            return;
        }

        Node<T> current = head;
        while (current.next.next != null) {
            current = current.next;
        }
        current.next = null;
    }

    public void deleteAtIndex(int index) {
        if (index < 0)
            return;

        if (index == 0) {
            head = head.next;
            return;
        }

        Node<T> current = head;
        Node<T> prev = null;
        int count = 0;

        while (current != null && count != index) {
            prev = current;
            current = current.next;
            count++;
        }

        if (current == null)
            return;

        prev.next = current.next;
    }

    public void replaceAtIndex(int index, T newValue) {
        if (index < 0)
            return;

        Node<T> current = head;
        int count = 0;

        while (current != null && count != index) {
            current = current.next;
            count++;
        }

        if (current != null) {
            current.data = newValue;
        }
    }

    public T get(int index) {
        if (index < 0 || head == null)
            throw new IndexOutOfBoundsException("Index out of bounds");

        Node<T> current = head;
        int count = 0;

        while (current != null && count != index) {
            current = current.next;
            count++;
        }

        if (current == null)
            throw new IndexOutOfBoundsException("Index out of bounds");

        return current.data;
    }

    public void printList() {
        Node<T> current = head;
        while (current != null) {
            System.out.print(current.data + " ");
            current = current.next;
        }
        System.out.println();
    }

    public void clear() {
        head = null;
    }
}

File: List/Flys/LinkedList.cs
using LinkedList;
using System;
using System.Collections.Generic;

public class LinkedList<T>
{
    public Node<T> Head { get; set; }

    public LinkedList(List<T> nodes)
    {
        this.Head = null;
        if (nodes != null && nodes.Count > 0)
        {
            Node<T> node = new Node<T>(nodes[0]);
            this.Head = node;
            for (int i = 1; i < nodes.Count; i++)
            {
                node.Next = new Node<T>(nodes[i]);
                node.Next.Prev = node;
                node = node.Next;
            }
        }
    }

    public void Append(T data)
    {
        if (this.Head == null)
        {
            this.Head = new Node<T>(data);
        }
        else
        {
            Node<T> node = this.Head;
            while (node.Next != null)
            {
                node = node.Next;
            }
            Node<T> newNode = new Node<T>(data);
            node.Next = newNode;
            newNode.Prev = node;
        }
    }

    public void Remove(T data)
    {
        Node<T> node = this.Head;
        while (node != null)
        {
            if (node.Data.Equals(data))
            {
                if (node.Prev != null)
                {
                    node.Prev.Next = node.Next;
                }
                else
                {
                    this.Head = node.Next;
                }
                if (node.Next != null)
                {
                    node.Next.Prev = node.Prev;
                }
                return;
            }
            node = node.Next;
        }
        throw new ArgumentException("Елемент не знайдено в списку");
    }

    public void Insert(int index, T data)
    {
        if (index < 0)
        {
            throw new IndexOutOfRangeException("Введіть невід'ємний індекс!");
        }
        else if (index == 0)
        {
            Node<T> newNode = new Node<T>(data);
            newNode.Next = this.Head;
            if (this.Head != null)
            {
                this.Head.Prev = newNode;
            }
            this.Head = newNode;
        }
        else
        {
            Node<T> node = this.Head;
            for (int i = 0; i < index - 1; i++)
            {
                if (node == null)
                {
                    throw new IndexOutOfRangeException("Невірний індекс");
                }
                node = node.Next;
            }
            if (node == null)
            {
                throw new IndexOutOfRangeException("Невірний індекс");
            }
            Node<T> newNode = new Node<T>(data);
            newNode.Next = node.Next;
            newNode.Prev = node;
            if (node.Next != null)
            {
                node.Next.Prev = newNode;
            }
            node.Next = newNode;
        }
    }

    public void Clear()
    {
        this.Head = null;
    }

    public List<T> Display()
    {
        List<T> elements = new List<T>();
        Node<T> node = this.Head;
        while (node != null)
        {
            elements.Add(node.Data);
            node = node.Next;
        }
        return elements;
    }
}
